### PR TITLE
Ista 1

### PR DESCRIPTION
**Pull request by EnLAI111**
I have added lamda_max (```reg_max``` in the code) to the algorithm ```ISTA```.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -37,21 +24,19 @@ This benchmark can be run using the following commands:
...
+	$ benchopt run benchmark_tv_1d -s solver1 -d dataset2 --max-runs 10 --n-repetitions 10
 
 
 Use `benchopt run -h` for more details about these options, or visit https://benchopt.github.io/api.html.
 
-.. |Build Template| image:: https://github.com/benchopt/template_benchmark/workflows/Tests/badge.svg
-   :target: https://github.com/benchopt/template_benchmark/actions
-.. |Build Status| image:: https://github.com/#ORG/#BENCHMARK_NAME/workflows/Tests/badge.svg
-   :target: https://github.com/#ORG/#BENCHMARK_NAME/actions
+.. |Build Status| image:: https://github.com/EnLAI111/benchmark_tv_1d/workflows/Tests/badge.svg
```
Comment (agramfort) [inline README.rst:39]:
```suggestion
.. |Build Status| image:: https://github.com/benchopt/benchmark_tv_1d/workflows/Tests/badge.svg
```

File: README.rst [hunk 2]
```diff
@@ -37,21 +24,19 @@ This benchmark can be run using the following commands:
 .. code-block::
 
    $ pip install -U benchopt
-   $ git clone https://github.com/#ORG/#BENCHMARK_NAME
-   $ benchopt run #BENCHMARK_NAME
+   $ git clone https://github.com/EnLAI111/benchmark_tv_1d
```
Comment (agramfort) [inline README.rst:27]:
```suggestion
   $ git clone https://github.com/benchopt/benchmark_tv_1d
```

File: README.rst [hunk 3]
```diff
@@ -37,21 +24,19 @@ This benchmark can be run using the following commands:
...
 
 
 Use `benchopt run -h` for more details about these options, or visit https://benchopt.github.io/api.html.
 
-.. |Build Template| image:: https://github.com/benchopt/template_benchmark/workflows/Tests/badge.svg
-   :target: https://github.com/benchopt/template_benchmark/actions
-.. |Build Status| image:: https://github.com/#ORG/#BENCHMARK_NAME/workflows/Tests/badge.svg
-   :target: https://github.com/#ORG/#BENCHMARK_NAME/actions
+.. |Build Status| image:: https://github.com/EnLAI111/benchmark_tv_1d/workflows/Tests/badge.svg
+   :target: https://github.com/EnLAI111/benchmark_tv_1d/actions
```
Comment (agramfort) [inline README.rst:40]:
```suggestion
   :target: https://github.com/benchopt/benchmark_tv_1d/actions
```

File: clean_template.py [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+    README = Path('README.rst')
+    text = README.read_text()
+
+    text = text.replace('#ORG', ORG)
+    text = text.replace('#BENCHMARK_NAME', BENCHMARK_NAME)
+
+    text = '\n'.join([line for line in text.splitlines()[13:]
+                      if 'template_benchmark' not in line] + [''])
+
+    README.write_text(text)
```
Comment (agramfort) [inline clean_template.py:23]: I think this file can be removed now

Comment (agramfort) [reply clean_template.py:23]: besides the .pyc file should not be commited

File: objective.py [hunk 5]
```diff
@@ -1,22 +1,22 @@
...
 class Objective(BaseObjective):
     name = "Ordinary Least Squares"
 
-    parameters = {
-        'fit_intercept': [False],
-    }
+    parameters = {'fit_intercept': [False]}
 
     def __init__(self, fit_intercept=False):
         self.fit_intercept = fit_intercept
```
Comment (agramfort) [inline objective.py:11]: I would remove the support of intercept for now

File: test_config.py [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
+
+
+def check_test_solver_install(solver_class):
+    """Hook called in `test_solver_install`.
+
+    If one solver needs to be skip/xfailed on some
+    particular architecture, call pytest.xfail when
+    detecting the situation.
+    """
+    pass
```
Comment (agramfort) [inline test_config.py:13]: TODO

Comment (tomMoral) [reply test_config.py:13]: This is not a todo :)
This needs to be modified only if a solver should not be tested.

Comment (EnLAI111) [reply test_config.py:13]: It is for checking if we have all the libraries and tools that we need in solver? Here for now I haven't put any special package in solver.

**Comment by agramfort** [comment]
@EnLAI111 you need to fix your flake8 errors

```
❯ flake8
./objective.py:9:5: E303 too many blank lines (2)
./objective.py:9:27: E251 unexpected spaces around keyword / parameter equals
./objective.py:9:29: E251 unexpected spaces around keyword / parameter equals
./datasets/simulated.py:13:16: E203 whitespace before ':'
./datasets/simulated.py:14:1: W191 indentation contains tabs
./datasets/simulated.py:14:1: E101 indentation contains mixed spaces and tabs
./datasets/simulated.py:14:6: E131 continuation line unaligned for hanging indent
./datasets/simulated.py:14:10: E203 whitespace before ':'
./datasets/simulated.py:25:1: W293 blank line contains whitespace
./datasets/simulated.py:28:1: W293 blank line contains whitespace
./datasets/simulated.py:30:1: W293 blank line contains whitespace
./solvers/python-ISTA.py:14:1: W293 blank line contains whitespace
./solvers/python-ISTA.py:20:1: W293 blank line contains whitespace
./solvers/python-ISTA.py:20:5: W191 indentation contains tabs
./solvers/python-ISTA.py:20:5: E101 indentation contains mixed spaces and tabs
./solvers/python-ISTA.py:34:1: W391 blank line at end of file
./solvers/python-ISTA.py:34:1: W293 blank line contains whitespace
```

**Review by agramfort — APPROVED** [review]
I would suggest to merge this PR quickly and then iterate with new solvers.

**Pull request merged**